server/routes/calendar.js

```diff
@@ -2,7 +2,42 @@ var express = require("express");
 var router = express.Router();
 
 var pool = require("../db/db");
-
+router.post("/changecoverage", (req, res) => {
+  var coverage = req.body.coverage;
+  approve = true;
+  var shiftID = req.body.shiftID;
+  var notes = req.body.sentNotes;
+  pool.query(
+    "UPDATE shifts SET cover_requested = $1 WHERE sle_id = $2",
+    [approve, shiftID],
+    (error, result) => {
+      if (error) {
+        //  throw error;
+      }
+    }
+  );
+  pool.query(
+    "SELECT sle_id FROM shifts WHERE shift_id = $1",
+    [shiftID],
+    (error, result) => {
+      if (error) {
+        throw error;
+      }
+      pool.query(
+        "INSERT INTO coverrequests (coverer_id, coveree_id, shift_id, supervisor_status, notes) VALUES ($3, null, $1, null, $2)",
```

Comment:
The coverer_id and coveree_id are switched here. The coverer_id is who is covering the shift, so it should be null and the coveree should have a value.

Reply:
resolved